packages/salesforcedx-apex-replay-debugger/src/core/heapDumpService.ts

```diff
@@ -336,36 +347,55 @@ export class HeapDumpService {
     ) {
       return varContainer.value;
     }
-    // For a list or set the name string is going to end up being (<val1>,...<valX)
-    // For a objects, the name string is going to end up being <TypeName>: {<Name1>=<Value1>,...<NameX>=<ValueX>}
-    const isListOrSet =
-      varContainer.type.toLowerCase().startsWith('list<') ||
-      varContainer.type.toLowerCase().startsWith('set<');
-    // The live debugger, for collections, doesn't include their type in variable name/value
-    const containerType = this.isCollectionType(varContainer.type)
-      ? ''
-      : `${varContainer.type}:`;
-    let returnString = isListOrSet ? '(' : `${containerType}{`;
-    let first = true;
-    // Loop through each of the container's variables to get the name/value combinations, calling to create
-    // the string if another collection is found.
-    for (const entry of Array.from(varContainer.variables.entries())) {
-      const valueAsApexVar = entry[1] as ApexVariableContainer;
-      if (!first) {
-        returnString += ', ';
-      }
-      if (valueAsApexVar.ref) {
-        // if this is also a ref then create the string from that
-        returnString += this.createStringFromVarContainer(valueAsApexVar);
+
+    // If the varContainer is a ref and it's already been visited then return the string
+    if (varContainer.ref) {
+      if (visitedMap.has(varContainer.ref)) {
+        return 'already output';
```

Comment:
I am guessing that this is the text matches the interactive debugger?

Reply:
Yes. The pictures in the description illustrate live VS replay which is why I'd used "already output"